diff --exclude='*.xml' --exclude='*.o.d' --exclude=.project --exclude=.cproject -Naur spandsp-base-0.0.6/src/t38_terminal.c spandsp-0.0.6/src/t38_terminal.c
--- spandsp-base-0.0.6/src/t38_terminal.c	2014-04-01 12:44:55.147810416 +0100
+++ spandsp-0.0.6/src/t38_terminal.c	2014-04-01 12:49:16.779085000 +0100
@@ -906,6 +906,11 @@
             fe->hdlc_tx.len = 0;
             front_end_status(s, T30_FRONT_END_SEND_STEP_COMPLETE);
             /* The above step should have got the next HDLC step ready - either another frame, or an instruction to stop transmission. */
+
+            /* Check if the process as been stopped */
+            if (s->t38_fe.timed_step == T38_TIMED_STEP_NONE)
+                return -1;
+
             if (fe->hdlc_tx.len < 0)
             {
                 /* End of transmission */
