diff --exclude='*.xml' --exclude='*.o.d' --exclude=.project --exclude=.cproject -Naur spandsp-base-0.0.6/src/spandsp/t30.h spandsp-0.0.6/src/spandsp/t30.h
--- spandsp-base-0.0.6/src/spandsp/t30.h	2010-08-09 15:12:29.000000000 +0100
+++ spandsp-0.0.6/src/spandsp/t30.h	2013-11-01 16:02:53.391710137 +0000
@@ -143,7 +143,9 @@
 /*! The maximum length of the body of an ident string */
 #define T30_MAX_IDENT_LEN           20
 /*! The maximum length of the user string to insert in page headers */
-#define T30_MAX_PAGE_HEADER_INFO    50
+#define T30_MAX_PAGE_HEADER_INFO    132
+/*! The maximum length of the user string to insert in a small page headers */
+#define T30_MAX_SMALL_PAGE_HEADER_INFO    50
 
 typedef struct t30_state_s t30_state_t;
 
diff --exclude='*.xml' --exclude='*.o.d' --exclude=.project --exclude=.cproject -Naur spandsp-base-0.0.6/src/t4_tx.c spandsp-0.0.6/src/t4_tx.c
--- spandsp-base-0.0.6/src/t4_tx.c	2010-07-24 17:28:35.000000000 +0100
+++ spandsp-0.0.6/src/t4_tx.c	2013-11-01 16:38:35.075637154 +0000
@@ -82,6 +82,7 @@
 #include "spandsp/timezone.h"
 #include "spandsp/t4_rx.h"
 #include "spandsp/t4_tx.h"
+#include "spandsp/t30.h"
 #if defined(SPANDSP_SUPPORT_T85)
 #include "spandsp/t81_t82_arith_coding.h"
 #include "spandsp/t85.h"
@@ -398,6 +399,42 @@
              s->current_page + 1);
 }
 /*- End of function --------------------------------------------------------*/
+static char *replace_str(char *str, const char *sub, char *rep)
+{
+    char buffer[T30_MAX_PAGE_HEADER_INFO + 1];
+    char *p;
+
+    if(! (p = strstr(str, sub)))  // icheck if 'sub' is in 'str'
+    {
+        return str;
+    }
+
+    strncpy(buffer, str, p - str); // copy chars 
+    buffer[p - str] = '\0';
+
+    snprintf(buffer + (p - str),T30_MAX_PAGE_HEADER_INFO,"%s%s", rep, p + strlen(sub));
+    strncpy(str, buffer, T30_MAX_PAGE_HEADER_INFO);
+    str[T30_MAX_PAGE_HEADER_INFO] = '\0';
+    return str;
+}
+
+static void make_header_full(t4_state_t *s, char *header)
+{
+    const char *sub_page = "%P";
+    const char *sub_far_ident = "%R";
+    char tmp[20];
+
+    strncpy(header,s->header_info,T30_MAX_PAGE_HEADER_INFO);  
+    header[T30_MAX_PAGE_HEADER_INFO] = '\0';
+    
+    snprintf(tmp,sizeof(tmp),"%d",s->current_page+1);
+    header = replace_str(header,sub_page,tmp);
+
+    snprintf(tmp,sizeof(tmp),"%s",(s->tiff.far_ident)  ?  s->tiff.far_ident  :  "");
+    header = replace_str(header,sub_far_ident,tmp);
+
+}
+/*- End of function --------------------------------------------------------*/
 
 static int t4_tx_put_fax_header(t4_state_t *s)
 {
@@ -407,10 +444,14 @@
     int pattern;
     int row_bufptr;
     char *t;
-    char header[132 + 1];
+    char header[T30_MAX_PAGE_HEADER_INFO + 1];
 
     /* Modify the resulting image to include a header line, typical of hardware FAX machines */
-    make_header(s, header);
+    
+    if (strlen(s->header_info) < T30_MAX_SMALL_PAGE_HEADER_INFO)
+       make_header(s, header);
+    else
+       make_header_full(s, header);
     switch (s->y_resolution)
     {
     case T4_Y_RESOLUTION_1200:
