diff --exclude='*.xml' --exclude='*.o.d' --exclude=.project --exclude=.cproject -Naur spandsp-base-0.0.6/src/t30.c spandsp-0.0.6/src/t30.c
--- spandsp-base-0.0.6/src/t30.c	2014-04-14 16:07:15.224777061 +0100
+++ spandsp-0.0.6/src/t30.c	2014-04-14 16:30:05.365280781 +0100
@@ -5866,6 +5866,11 @@
         case T30_STATE_B:
             /* We have now allowed time for the last message to flush through
                the system, so it is safe to report the end of the call. */
+
+            /* If we have already have received the EOP ignore DCN retry and call dropped */
+            if (s->next_rx_step == T30_EOP && (s->current_status == T30_ERR_RETRYDCN || s->current_status == T30_ERR_CALLDROPPED))
+               s->current_status = T30_ERR_OK;
+
             if (s->phase_e_handler)
                 s->phase_e_handler(s, s->phase_e_user_data, s->current_status);
             set_state(s, T30_STATE_CALL_FINISHED);
@@ -6108,6 +6113,8 @@
 
 SPAN_DECLARE(void) t30_terminate(t30_state_t *s)
 {
+    span_log(&s->logging, SPAN_LOG_FLOW, "Enter terminate, next step - %d | state: %d | phase: %d\n", s->next_rx_step, s->state, s->phase);
+
     if (s->phase != T30_PHASE_CALL_FINISHED)
     {
         /* The far end disconnected early, but was it just a tiny bit too early,
@@ -6127,12 +6134,19 @@
             s->current_status = T30_ERR_CALLDROPPED;
             break;
         }
+
+        /* If we have already have received the EOP ignore DCN retry and call dropped */
+        if (s->next_rx_step == T30_EOP && (s->current_status == T30_ERR_RETRYDCN || s->current_status == T30_ERR_CALLDROPPED))
+           s->current_status = T30_ERR_OK;
+
         if (s->phase_e_handler)
             s->phase_e_handler(s, s->phase_e_user_data, s->current_status);
         set_state(s, T30_STATE_CALL_FINISHED);
         set_phase(s, T30_PHASE_CALL_FINISHED);
         release_resources(s);
     }
+
+    span_log(&s->logging, SPAN_LOG_FLOW, "Exit terminate, next step - %d | state: %d | phase: %d\n", s->next_rx_step, s->state, s->phase);
 }
 /*- End of function --------------------------------------------------------*/
 
